simulator/modules/decode/decode.cpp

```diff
@@ -87,6 +96,24 @@ void Decode<FuncInstr>::clock( Cycle cycle)
 
     auto instr = read_instr( cycle);
 
+    /* acquiring real information for BPU */
+    wp_bp_update->write( instr.get_bp_upd(), cycle);
+
+    bool is_misprediction = instr.is_direct_jump() &&
+    ( instr.get_bp_data().is_taken != instr.is_taken() || instr.get_bp_data().target != instr.get_decoded_target());
```

Comment:
> instr.get_bp_data().is_taken != instr.is_taken()

It has to be

> !instr.get_bp_data().is_taken

since `instr.is_taken()` always true for direct jumps.